.github/workflows/lint.yaml

```diff
@@ -0,0 +1,27 @@
+name: Lint source code
+on:
+  - push
+  - workflow_dispatch
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Update no-OS submodule
+        run: |
+          git submodule update --init libraries/no-OS
```

Comment:
why do we need to explicitly checkout libraries/no-OS? My understanding is style checks are performed only on application source files present on this repo and we don't need any other libraries or 3rd part repos for this. Is that correct?

Reply:
Ah, nevermind. This is necessary because it is using the astyle configuration files from no-OS below.